### PR TITLE
All rules include ones with bugs

### PR DESCRIPTION
**Pull request by nivcertora**
Fix uniqueSigner and createAndGetSignerEquivalence rules, Add assumptions for address size and bytes word alignment, 
Also fix verifySignatureEq using summaries for encodeDataJson and verifyAllowMeallbility 

WebAuth: https://prover.certora.com/output/1512/15e25323c0a54682810ddc5c3a4f6ccc?anonymousKey=c38ad8f1ce989fff56794a963eb688b6f6807f3f
GetSigner: 
https://prover.certora.com/output/1512/a7d4f9ef71b24612ad52a5a9c4196f90?anonymousKey=34bfebe8bc2787ffbd2c3937ed44c192d55ce762
Factrory:
https://prover.certora.com/output/1512/49143cd8f8744ca38faae389e6b17f7a?anonymousKey=29fc317ca578d6ac65686b9e489da3f41bc01734

### CODE REVIEW

File: certora/harnesses/WebAuthnHarness.sol [hunk 1]
```diff
@@ -4,6 +4,23 @@ pragma solidity ^0.8.0;
 import {P256, WebAuthn} from "../../modules/passkey/contracts/libraries/WebAuthn.sol";
 
 contract WebAuthnHarness {
+
+    mapping (bytes32 => mapping (bytes32 => string))  symbolicClientDataJson;
+
+    function SencodeDataJson(bytes32 challenge, string calldata clientDataFields) public returns (string memory){
```
Comment (yoav-el-certora) [inline certora/harnesses/WebAuthnHarness.sol]: Typo? **S**encode....

File: certora/specs/WebAuthn.spec [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
+    env e;
+    return SencodeDataJson(e, challenge, clientDataFields);
+}
 
-    string a1 = encodeClientDataJson(e, challenge1, clientDataFields);
-    string b1 = encodeClientDataJson(e, challenge2, clientDataFields);
+ghost checkInjectiveSummary(bytes32, bytes32, bytes32) returns bool {
+    axiom forall bytes32 x1. forall bytes32 y1. forall bytes32 x2. forall bytes32 y2. forall bytes32 result.
+    (x1 != x2) => !(checkInjectiveSummary(x1, y1, result) && checkInjectiveSummary(x2, y2, result));
+}
```
Comment (yoav-el-certora) [inline certora/specs/WebAuthn.spec]: We updated this ghost -
from (a != b) => !(c && d)
to      c && d => a ==b
which is logically equal.

File: modules/passkey/contracts/SafeWebAuthnSignerFactory.sol [hunk 3]
```diff
@@ -32,7 +32,8 @@ contract SafeWebAuthnSignerFactory is ISafeSignerFactory {
     /**
      * @inheritdoc ISafeSignerFactory
      */
-    function getSigner(uint256 x, uint256 y, P256.Verifiers verifiers) public view override returns (address signer) {
```
Comment (yoav-el-certora) [inline modules/passkey/contracts/SafeWebAuthnSignerFactory.sol]: I don't think its a good idea to edit safe source code.
Why wont we just add a munged file to replace the whole file with the harness addition?

File: certora/harnesses/WebAuthnHarness.sol (hunk 1 above)
Comment (nivcertora) [reply certora/harnesses/WebAuthnHarness.sol]: Renamed to summaryEncodeDataJson to make it clear

File: certora/specs/WebAuthn.spec (hunk 2 above)
Comment (nivcertora) [reply certora/specs/WebAuthn.spec]: Thanks, changed to : 
(checkInjectiveSummary(x1, y1, result) && checkInjectiveSummary(x2, y2, result)) => (x1 == x2);

File: modules/passkey/contracts/SafeWebAuthnSignerFactory.sol (hunk 3 above)
Comment (nivcertora) [reply modules/passkey/contracts/SafeWebAuthnSignerFactory.sol]: Removed changes from source code and move to munged

**Pull request merged**